lib/stemcell/launcher.rb

```diff
@@ -61,37 +62,25 @@ class Launcher
     TEMPLATE_PATH = '../templates/bootstrap.sh.erb'
     LAST_BOOTSTRAP_LINE = "Stemcell bootstrap finished successfully!"
 
+    MAX_RUNNING_STATE_WAIT_TIME = 300 # seconds
+    RUNNING_STATE_WAIT_SLEEP_TIME = 5 # seconds
+
     def initialize(opts={})
       @log = Logger.new(STDOUT)
       @log.level = Logger::INFO unless ENV['DEBUG']
       @log.debug "creating new stemcell object"
       @log.debug "opts are #{opts.inspect}"
 
-      REQUIRED_OPTIONS.each do |req|
-        raise ArgumentError, "missing required param #{req}" unless opts[req]
-        instance_variable_set("@#{req}",opts[req])
+      REQUIRED_OPTIONS.each do |opt|
+        raise ArgumentError, "missing required option 'region'" unless opts[opt]
```

Comment:
You should probably not hardcode "region" in the error string.

P.S. No, I don't know why I'm reading this.